src/pydvl/value/result.py

```diff
@@ -0,0 +1,580 @@
+"""
+This module collects types and methods for the inspection of the results of
+valuation algorithms.
+
+The most important class is :class:`ValuationResult`, which provides access
+to raw values, as well as convenient behaviour as a ``Sequence`` with extended
+indexing and updating abilities, and conversion to `pandas DataFrames
+<https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html>`_.
+"""
+import collections.abc
+import logging
+from dataclasses import dataclass
+from functools import total_ordering
+from numbers import Integral
+from typing import (
+    Any,
+    Generator,
+    Iterable,
+    List,
+    Literal,
+    Optional,
+    Sequence,
+    Union,
+    cast,
+    overload,
+)
+
+import numpy as np
+from numpy.typing import NDArray
+
+from pydvl.utils.dataset import Dataset
+from pydvl.utils.numeric import running_moments
+from pydvl.utils.status import Status
+
+try:
+    import pandas  # Try to import here for the benefit of mypy
+except ImportError:
+    pass
+
+__all__ = ["ValuationResult"]
+
+logger = logging.getLogger(__name__)
+
+
+@total_ordering
+@dataclass
+class ValueItem:
+    """The result of a value computation for one datum.
+
+    ValueItems can be compared with the usual operators. These take only the
+    :attribute:`value` into account
+
+    .. todo::
+       Maybe have a mode of comparing similar to `np.isclose`, or taking the
+       :attribute:`variance` into account.
+    """
+
+    #: Index of the sample with this value in the original :class:`Dataset`
+    index: int
+    #: Name of the sample if it was provided. Otherwise, `str(index)`
+    name: str
+    #: The value
+    value: float
+    #: Variance of the value if it was computed with an approximate method
+    variance: Optional[float]
+    #: Number of updates for this value
+    count: Optional[int]
+
+    def __lt__(self, other):
+        return self.value < other.value
+
+    def __eq__(self, other):
+        return self.value == other.value
+
+    def __index__(self) -> int:
+        return self.index
+
+    @property
+    def stderr(self) -> Optional[float]:
+        """Standard error of the value."""
+        if self.variance is None or self.count is None:
+            return None
+        return float(np.sqrt(self.variance / self.count).item())
+
+
+class ValuationResult(collections.abc.Sequence):
```

Comment:
Stupid git move and diffs don't go well together, sorry. Key points here are:
* sorting is simplified and hopefully indexing more consistent and pandas-like (and yes, I've finally realized that this object would be much simpler with a DataFrame as backend 🤦🏽)
* it's possible to sort by different keys, but I decided to deviate a bit from the standard interface for simplicity (sorry)
* default indexing of all attributes is based on data indices
* ipos-equivalent indexing is done with a dedicated method `get` (maybe I could rename that).
* Addition of results is also more stable and generalised to any data indices, making the `or` operator superfluous. This makes aggregating parallel results much simpler (just add them and it always works)

Reply:
This class is becoming really big... Can't most of this be done using dataframes?

Reply:
Yes it can, and it would be much easier and robust and flexible, see my comment. But I don't have the energy to rewrite everything _again_... If it works, let's not touch it, until we need some other feature and someone feels like redoing all of it.